### PR TITLE
Support nested arrays without losing the structure

### PR DESCRIPTION
**Pull request by ramirezalejo**
I have been trying to return a nested array from a custom method, but on return the nested array was converted to a single dimension array, with this change the nested arrays are returned in the same way the custom method returned it.

### CODE REVIEW

**Comment by ramirezalejo** [comment]
I added recursively from the second dimension, it is because in the cases when I returned three or more dimensions as the hole result, the array declaration from the transformer.json creates an additional unnecessary dimension. Thanks for your comments.

**Comment by Courela** [comment]
Add some Unit Tests if you can

**Comment by ramirezalejo** [comment]
I just added a test case for the scenario, I added only one because the code will be reached only in cases when a custom method returns nested arrays. Looking forward to see if you have any comments.

File: Examples/Input_Customfunction_Nestedresult.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
+﻿{
```
Comment (Courela) [inline Examples/Input_Customfunction_Nestedresult.json]: This file will be better off in folder UnitTests/Inputs, there's already some input files there, and it avoids the ugly path in CustomFunctionsTest file.

Comment (ramirezalejo) [reply Examples/Input_Customfunction_Nestedresult.json]: Got it, I just moved to Inputs in tests the files and changed the paths.

**Comment by davorjhub** [comment]
@Courela @WorkMaze  When will this feature be released, i have waiting for this feature for so long?

**Comment by Courela** [comment]
Not really in my power, I'm just a contributor.
If you need it badly you can use @ramirezalejo repository and compile from there.

**Pull request merged**